AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift

```diff
@@ -30,6 +30,7 @@ public class FullscreenMessage: NSObject, WKNavigationDelegate, FullscreenPresen
     var listener: FullscreenMessageDelegate?
     public private(set) var webView: UIView?
     private var messageMonitor: MessageMonitoring
+    private var loadingNavigation: WKNavigation?
```

Comment:
This is an interesting approach... typically I've seen it done like:

1. Add the WKNavigationDelegate protocol to your class declaration, like: `class FullscreenMessage: WKNavigationDelegate`
2. Set the `navigationDelegate` property of `webView` to `self,` before loading the web request, like: `webView?.navigationDelegate = self`
3. Then implement `webView(_:didFinish:)`

The approach you've laid out should be fine but isn't one I've seen used before.

Reply:
Isn't this current approach the same that you have mentioned ?
The only extra thing that's is happening is we filter out all the other page loads from `webView(_:didFinish:)`

setting the loadNavigation

Reply:
Good point, I didn't notice the class conforms to `WKNavigationDelegate`.